backend/src/game_engine/services/itemServece.ts

```diff
@@ -0,0 +1,59 @@
+import Matter from 'matter-js';
+
+import * as Constants from '../../constants/constants';
+import GameEngine from '../../rooms/GameEngine';
+import Item from '../../rooms/schema/Item';
+
+// eslint-disable-next-line @typescript-eslint/no-extraneous-class
+export default class ItemService {
+  private readonly gameEngine: GameEngine;
+
+  constructor(gameEngine: GameEngine) {
+    this.gameEngine = gameEngine;
+  }
+
+  createRandomItems(rows: number, cols: number) {
+    const items = [];
+    for (let x = 2; x < cols - 1; x += 2) {
+      for (let y = 2; y < rows; y += 2) {
+        const index = Math.floor(Math.random() * 6);
+        if (index === 1)
+          items.push(
+            this.addItem(
+              this.gameEngine.state.createItem(x, y, Constants.ITEM_TYPE.BOMB_POSSESSION_UP)
+            )
+          );
+        else if (index === 2)
+          items.push(
+            this.addItem(this.gameEngine.state.createItem(x, y, Constants.ITEM_TYPE.BOMB_STRENGTH))
+          );
+        else if (index === 3)
+          items.push(
+            this.addItem(this.gameEngine.state.createItem(x, y, Constants.ITEM_TYPE.PLAYER_SPEED))
+          );
+      }
+    }
+    Matter.Composite.add(this.gameEngine.world, items);
+  }
+
+  addItem(item: Item) {
+    const itemBody = Matter.Bodies.rectangle(
+      item.x,
+      item.y,
+      Constants.TILE_WIDTH,
+      Constants.TILE_HEIGHT,
+      {
+        label: item.itemType,
+      }
+    );
+    this.gameEngine.itemBodies.set(item.id, itemBody);
+    return itemBody;
+  }
+
+  removeItem(item: Item) {
```

Comment:
Item Service の中なので add / remove でわかるかなと思いました（といいつつ bomb で自分もやっちゃってますが・・）

```suggestion
  add(item: Item) {
    const itemBody = Matter.Bodies.rectangle(
      item.x,
      item.y,
      Constants.TILE_WIDTH,
      Constants.TILE_HEIGHT,
      {
        label: item.itemType,
      }
    );
    this.gameEngine.itemBodies.set(item.id, itemBody);
    return itemBody;
  }

  remove(item: Item) {
```

Reply:
addはmatter.compositeやphaserの方でも使われているので、addだけでは混乱してしまうかな～と思うのでこのままでも良いのかなとも感じています。
addを呼び出しているオブジェクトを見れば問題ない話ですが、、